pmda/parallel.py

```diff
@@ -313,26 +313,31 @@ def run(self,
         start, stop, step = self._trajectory.check_slice_indices(
             start, stop, step)
         n_frames = len(range(start, stop, step))
-        bsize = int(np.ceil(n_frames / float(n_blocks)))
+        idx = make_balanced_blocks(n_frames, n_blocks,
+                                   start=start, step=step)
 
         with timeit() as total:
             with timeit() as prepare:
                 self._prepare()
             time_prepare = prepare.elapsed
             blocks = []
             with self.readonly_attributes():
-                for b in range(n_blocks):
+                for bstart, bstop in zip(idx[:-1], idx[1:]):
                     task = delayed(
                         self._dask_helper, pure=False)(
-                            b * bsize * step + start,
-                            min(stop, (b + 1) * bsize * step + start),
+                            bstart,
+                            min(bstop, stop),
                             step,
                             self._indices,
                             self._top,
                             self._traj, )
                     blocks.append(task)
                 blocks = delayed(blocks)
                 res = blocks.compute(**scheduler_kwargs)
+            # hack to handle n_frames == 0 in this framework
```

Comment:
Why not handle this special case in `__init__` and throw an error? At least a warning would be nice.

Reply:
Added warning + test in 61d087d374a5718ef1d06ff2485e4dc159b25bfa